lib/improver/tests/ensemble_calibration/ensemble_calibration/helper_functions.py

```diff
@@ -38,9 +38,85 @@
 import iris
 from iris.coords import AuxCoord, DimCoord
 from iris.cube import Cube, CubeList
+from iris.tests import IrisTest
 import numpy as np
 
+from improver.tests.set_up_test_cubes import set_up_variable_cube
 from improver.utilities.cube_manipulation import concatenate_cubes
+from improver.utilities.warnings_handler import ManageWarnings
+
+
+IGNORED_MESSAGES = ["Collapsing a non-contiguous coordinate.",
+                    "Not importing directory .*sphinxcontrib'",
+                    "The pandas.core.datetools module is deprecated",
+                    "numpy.dtype size changed",
+                    "The statsmodels can not be imported",
+                    "invalid escape sequence",
+                    "can't resolve package from",
+                    "Minimisation did not result in"
+                    " convergence",
+                    "\nThe final iteration resulted in a percentage "
+                    "change that is greater than the"
+                    " accepted threshold ",
+                    "divide by zero encountered in true_divide",
+                    "invalid value encountered in"]
+WARNING_TYPES = [UserWarning, ImportWarning, FutureWarning, RuntimeWarning,
+                 ImportWarning, DeprecationWarning, ImportWarning, UserWarning,
+                 UserWarning, RuntimeWarning, RuntimeWarning]
+
+
+class SetupCubes(IrisTest):
+
+    """Set up cubes for testing."""
+
+    @ManageWarnings(
+        ignored_messages=IGNORED_MESSAGES, warning_types=WARNING_TYPES)
+    def setUp(self):
+        """Set up temperature and wind speed cubes for testing."""
+        # Note: test_temperature_realizations_data_check produces ~0.5K
+        # different results when the temperature forecast cube is float32
```

Comment:
Iterative minimisation schemes can be very sensitive to the precision, especially when you get close to the minimum and the lambda is incredibly small - I'd say that it's probably safest/best to use `float64` for this algorithm, then just recast the final results to `float32` - at least until we've done some kind of sensitivity study.
(@benfitzpatrick for interest)

Reply:
I don't have a problem with this, if it's isolated away from the rest of the code

Reply:
I've updated the inputs to the minimisation to be float64 precision and this seems to give more stable solutions across different machines, at least on the machines that I've tested on. The resulting coefficients are set to float32, so I don't think that there should be any more general precision escalation as a result of this.